portal-impl/src/com/liferay/portal/servlet/ComboServlet.java

```diff
@@ -307,7 +307,7 @@ protected void doService(
 			if (cacheEnabled && (modulePathsString != null) &&
```

Comment:
If we enforce the property at this point, we are leaving out the case where cache is not enabled. The enforcement must take place no matter wether we want to cache the response or not.

Indeed, if we fail the response due to request exceeds `PropsValues.COMBO_MAX_FILES` then combo must not cache such response either.